examples/twittermap/web/public/javascripts/common/services.js

```diff
@@ -225,6 +235,31 @@ angular.module('cloudberry.common', ['cloudberry.mapresultcache'])
       };
     }
 
+    function handleByTimeRequest(parameters, geoIds) {
```

Comment:
In this function, we don't need a batch request at all. A single query can also specify an `option` attribute:
```
{
          dataset: parameters.dataset,
          filter: getFilter(parameters, defaultNonSamplingDayRange, parameters.geoIds),
          group: {
            by: [{
              field: "create_at",
              apply: {
                name: "interval",
                args: {
                  unit: parameters.timeBin
                }
              },
              as: parameters.timeBin
            }],
            aggregate: [{
              field: "*",
              apply: {
                name: "count"
              },
              as: "count"
            }]
          },
          option: {
            sliceMillis: cloudberryConfig.querySliceMills
          }
        }
```
And for in different cases, you may need to add this attribute dynamically, you can always manipulate an object in Javascript like this:
```
// Suppose query is previously constructed JSON object
// Add "transform" attribute to the query JSON
        query["transform"] = {
          wrap: {
            id: id ? id : category,
            category: category
          }
        };
```